index.js

```diff
@@ -46,244 +42,248 @@ class AntiSpam extends Events.EventEmitter {
 					);
-				await msg.channel
-					.send(
-						`Could not kick **${msg.author.tag}** because of inpropper permissions.`
-					)
-					.catch((e) => {
-						if (options.verbose) console.error(e);
-					});
 				return false;
 			}
 
 			try {
-				await msg.member.kick('Spamming!');
-				this.emit('kickAdd', msg.member);
-			} catch (e) {
+				await message.member.kick('Spamming!');
+				if (options.kickMessage)
+					await message.channel
+						.send(format(options.kickMessage, message))
+						.catch(e => {
+							if (options.verbose) console.error(e);
+						});
+				this.emit('kickAdd', message.member);
+				return true;
+			} catch (error) {
+				const emitted = this.emit('error', message, error, 'kick');
+				if (emitted) return false;
 				if (options.verbose)
 					console.log(
-						`**${msg.author.tag}** (ID: ${msg.author.id}) could not be kicked, ${e}.`
+						`**${message.author.tag}** (ID: ${message.author.id}) could not be kicked, ${error}.`
 					);
-				await msg.channel
+				await message.channel
 					.send(
-						`Could not kick **${msg.author.tag}** because of an error: \`${e}\`.`
+						`Could not kick **${message.author.tag}** because of an error: \`${error}\`.`
 					)
-					.catch((e) => {
+					.catch(e => {
 						if (options.verbose) console.error(e);
 					});
 				return false;
 			}
-
-			let msgToSend = formatString(options.kickMessage, msg);
-
-			await msg.channel.send(msgToSend).catch((e) => {
-				if (options.verbose) console.error(e);
-			});
-			return true;
 		};
 
-		const warnUser = async (msg) => {
-			// Mark the user as warned
-			warnedUsers.push(msg.author.id);
+		const warnUser = async () => {
+			data.warnedUsers.push(message.author.id);
 			this.emit('warnAdd', message.member);
 
-			let msgToSend = formatString(options.warnMessage, msg);
-
-			await msg.channel.send(msgToSend).catch((e) => {
-				if (options.verbose) console.error(e);
-			});
+			if (options.warnMessage)
+				await message.channel
+					.send(format(options.warnMessage, message))
+					.catch(e => {
+						if (options.verbose) console.error(e);
+					});
 
 			return true;
 		};
 
-		users.push({
+		data.users.push({
 			time: Date.now(),
-			author: message.author.id
+			id: message.author.id
 		});
 
-		messageCache.push({
+		data.messageCache.push({
 			content: message.content,
 			author: message.author.id
 		});
 
-		let messageMatches = messageCache.filter(
-			(m) => m.content === message.content && m.author === message.author.id
+		const messageMatches = data.messageCache.filter(
+			m => m.content === message.content && m.author === message.author.id
 		).length;
-		let spamMatches = users.filter(
-			(u) =>
-				u.time > Date.now() - options.maxInterval &&
-				u.author === message.author.id
+		const spamMatches = data.users.filter(
+			u =>
+				u.time > Date.now() - options.maxInterval && u.id === message.author.id
```

Comment:
I think we should use a unique line, as lines 204-205

Reply:
wdym? lines 204-205? or the filter function all on one line?

Reply:
lines 204-205 format:
```js
m => m.content === message.content && m.author === message.author.id
```
lines 208-209 format:
```js
u =>
     u.time > Date.now() - options.maxInterval && u.id === message.author.id
```
In all case, we need to use the same format everywhere and I think the first (204-205) is better.

Reply:
ight, updated the formatting a little, hows that looking?